docroot/modules/custom/sitenow_dispatch/src/Form/SettingsForm.php

```diff
@@ -55,42 +55,155 @@ public static function create(ContainerInterface $container) {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->configFactory()->get('sitenow_dispatch.settings');
+    $form['#tree'] = TRUE;
+
     $form['description_text'] = [
       '#markup' => '<p><a href="https://its.uiowa.edu/dispatch">Dispatch</a> is a web service that allows users to create and manage campaigns to generate PDF content, email messages, SMS messages, or voice calls. You must have a Dispatch <a href="https://apps.its.uiowa.edu/dispatch/help/faq#q1">client and account</a> to use Dispatch functionality within your site.</p>',
     ];
 
     $form['api_key'] = [
-      '#type' => 'textfield',
+      '#type' => 'password',
+      '#required' => TRUE,
       '#title' => $this->t('API key'),
-      '#default_value' => $this->config('sitenow_dispatch.settings')->get('api_key'),
+      '#attributes' => [
+        'value' => $config->get('api_key') ?? NULL,
+      ],
       '#description' => $this->t('A valid Dispatch client API key. See the Dispatch <a href="https://apps.its.uiowa.edu/dispatch/help/api">API key documentation</a> for more information.'),
     ];
-    return parent::buildForm($form, $form_state);
-  }
 
-  /**
-   * {@inheritdoc}
-   */
-  public function submitForm(array &$form, FormStateInterface $form_state) {
-    $this->config('sitenow_dispatch.settings')
-      ->set('api_key', $form_state->getValue('api_key'))
-      ->save();
-    parent::submitForm($form, $form_state);
+    if ($client = $config->get('client')) {
+      $form['api_key']['#description'] .= $this->t(' <em>Currently set to @client client</em>.', [
+        '@client' => $client,
+      ]);
+    }
+
+    // Grab the current user to set access to the Thanks
+    // form settings only for administrators.
+    $current_user = $this->currentUser();
+    $form['thanks'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Thank You Email'),
+      '#description' => $this->t('Thank You email configuration. The <a href="@link">1row x 1col curated Dispatch template</a> is used.', [
+        '@link' => 'https://apps.its.uiowa.edu/dispatch/help/curatedtemplate/UI%201%20row%20x%201%20col%20-%20Version%202',
+      ]),
+      '#access' => in_array('administrator', $current_user->getRoles()),
```

Comment:
The user 1 account may not have the administrator role. There is a service in uiowa_core that accounts for this: https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/uiowa_core/src/Access/UiowaCoreAccess.php

Here is a hook-based example of using it: https://github.com/uiowa/uiowa/blob/d8e08172bf66c74ada567ee5e2c9c887c3b34224/docroot/modules/custom/uiowa_aggregator/uiowa_aggregator.module#L29-L38

Reply:
I think `$access->isAllowed()` would work here.

Reply:
Cool! Thanks for the example--got it updated.